include/video.h

```diff
@@ -188,18 +188,18 @@ enum class PixelFormat : uint8_t {
 	// FFmpeg Equivalent: AV_PIX_FMT_BGR24
 	BGR24_ByteArray = 24,
 
-	// 16.7M (32-bit) true colour; 8 bits per red/blue/green component;
-	// stored as packed uint32 data with highest 8 bits unused
+	// 16.7M (32-bit) true colour, 8 bits per red/blue/green component;
+	// stored as packed 32-bit data
 	//
-	// Stored as array of uint32_t in host native endianess.
-	// Each uint32_t is layed out as follows:
-	// (msb)8X 8R 8G 8B(lsb)
+	// Stored as array of uint8_t in BGRA memory order (endian agnostic)
 	// Example:
-	// uint32_t pixel = (unused << 24) | (red << 16) | (green << 8) | (blue << 0)
+	// uint8_t *pixels = image.image_data;
+	// pixels[0] = blue; pixels[1] = green; pixels[2] = red; pixels[3] = alpha;
 	//
-	// SDL Equivalent: SDL_PIXELFORMAT_XRGB8888
-	// FFmpeg Equivalent: AV_PIX_FMT_0RGB32
-	XRGB8888_Packed32 = 32
+	// SDL Equivalent: SDL_PIXELFORMAT_BGRA32 (alias for SDL_PIXELFORMAT_BGR24)
```

Comment:
Yes, those two are not aliases for each other.  `SDL_PIXELFORMAT_BGRA32` aliases to either `SDL_PIXELFORMAT_BGRA8888` or `SDL_PIXELFORMAT_ARGB8888` depending on endianess.  These are 4 bytes per pixel,

`SDL_PIXELFORMAT_BGR24 ` is 3 bytes per pixel stored in a byte array.

`SDL_PIXELFORMAT_BGRX32` does not exist in SDL.  You would have to write your own marcro to alias to either `SDL_PIXELFORMAT_BGRX8888` or `SDL_PIXELFORMAT_XRGB8888`  depending on endianess.  I believe SDL internally always loads 32 bit formats as 32 bit packed words so an alias is needed if what you're describing is a byte array.

https://github.com/libsdl-org/SDL/blob/15ead9a40d09a1eb9972215cceac2bf29c9b77f6/include/SDL_pixels.h#L273-L284

I wrote these comments by the way.  @johnnovak and I went back and forth for quite a while on how to best describe these pixel formats.  It can be quite confusing and different libraries/vendors have different ways of describing them.

Reply:
1. Yeah, what @weirddan455 writes is true, and I remember this confused me quite a bit too about the SDL pixel formats @kcgen. They're simply a bit sloppy in their pixel format namings, that's the gist of it, and as @weirddan455 said, there's no out-of-the-box SDL pixel format definition for `BGRX32_ByteArray`. It would be best to drop the SDL equivalent line for this altogether.

2. As stated, `BGRX32_ByteArray` is the correct name to use here. I'm 99% sure we just have random garbage in the 4th byte, or maybe all zeroes. Based on your fix to the image decoder in this PR, I'm confident you got it right and this is indeed 4-components as a byte array 😄 

3. "OpenGL Equivalent: GL_BGRA with data type GL_UNSIGNED_INT_8_8_8_8_REV" — that's probably incorrect. `GL_BGRA` just specifies the "register layout", and `GL_UNSIGNED_INT_8_8_8_8_REV` just specifies the memory layout, plus tells the driver to reverse the bytes. You need to use the `*_REV` or non-REV version (`GL_UNSIGNED_INT_8_8_8_8`) depending on whether you're on a BE or LE architecture.

    What you really want is this ([source](https://afrantzis.com/pixel-format-guide/opengl.html)):

    ```
    Example: GL_RGBA with GL_UNSIGNED_BYTE

    This combination specifies a pixel format laid out in memory as the bytes R, G, B, A
    (R at the lowest address, A at the highest)
    ```

    If you really want to include the OpenGL "equivalent", make sure to mention this parenthesis: "(assuming the alpha component is set to 255; what we have here is really `GL_RGBX`, so the 4th byte is unused/undfined, but that has no OpenGL equivalent)".

Reply:
(Removed references to SDL, OpenGL, and ffmpeg equivalent names.)